plugin/path_checkout_sets.go

```diff
@@ -21,6 +21,20 @@ type librarySet struct {
 	DisableCheckInEnforcement bool          `json:"disable_check_in_enforcement"`
 }
 
+// Validates ensures that a set meets our code assumptions that TTLs are set in
+// a way that makes sense, and that there's at least one service account.
+func (l *librarySet) Validate() error {
```

Comment:
Does this also need to check that `MaxTTL` and `TTL` are >=0, or is that handled elsewhere?

Reply:
Couldn't they both be 0 to indicate an infinite checkout?

Reply:
Well, and also I tested by hand that way and I found that the checkout defaulted to, essentially, 32 days due to backend-level default TTLs.

Reply:
I’m referring to checking for negative TTLs.

Reply:
Oh! I gotcha. Good question. That's handled at a framework level because of it being a `TypeDurationSecond`.
```
$ vault write ad/library/accounting-team \
>     service_account_names=fizz@example.com,buzz@example.com \
>     ttl=-1000 \
>     max_ttl=-1000
Error writing data to ad/library/accounting-team: Error making API request.

URL: PUT http://localhost:8200/v1/ad/library/accounting-team
Code: 400. Errors:

* error converting input -1000 for field "max_ttl": cannot provide negative value '-1000'
```
```
$ vault write ad/library/accounting-team     service_account_names=fizz@example.com,buzz@example.com     ttl=-1000
Error writing data to ad/library/accounting-team: Error making API request.

URL: PUT http://localhost:8200/v1/ad/library/accounting-team
Code: 400. Errors:

* error converting input -1000 for field "ttl": cannot provide negative value '-1000'
```

Reply:
👍